### PR TITLE
Fix catch-all exceptions for export

### LINKED ISSUE #80: missing argument error as user exception
Missing argument in constructor ended up as a user exception and it flew under our radar.

http://status.keboola.com/increased-error-rate-in-mysql-extractor

### PR DESCRIPTION
**Pull request by pivnicek**
fixes #80

### CODE REVIEW

**Comment by odinuv** [comment]
i'd add tests for this - something like writing to readonly file (which should not happen), with logger TestHandler we can check that no retry was done

File: tests/RetryTest.php [hunk 1]
```diff
@@ -202,4 +206,22 @@ public function testDeadConnectionException(): void
...
+    {
+        $testLogger = new Logger('common-retry-test-logger');
+        $testLogger->pushHandler(new TestHandler());
+
+        $config = $this->getRetryConfig();
+        @unlink($this->dataDir . '/out/tables/in.c-main.sales.csv');
+        touch($this->dataDir . '/out/tables/in.c-main.sales.csv');
+        chmod($this->dataDir . '/out/tables/in.c-main.sales.csv', 444);
+
+        try {
```
Comment (odinuv) [inline tests/RetryTest.php]: i'm not sure about phpunit version here, but if self::expectException and self::expectExceptionMessage are available better use them

**Comment by pivnicek** [comment]
ok @odinuv so this tests the catch CsvException (which is good), but doesn't test the fall through.

File: tests/CommonExtractorTest.php [hunk 2]
```diff
@@ -930,6 +931,19 @@ public function testInvalidConfigsInvalidTableWithNoName(): void
         ($this->getApp($config))->run();
     }
 
+    public function testNoRetryOnCsvError(): void
+    {
+        $config = $this->getConfigRowForCsvErr(self::DRIVER);
+
+        (new Filesystem)->remove($this->dataDir . '/out/tables/in.c-main.simple-csv-err.csv');
+        (new Filesystem)->symlink('/dev/full', $this->dataDir . '/out/tables/in.c-main.simple-csv-err.csv');
```
Comment (pivnicek) [inline tests/CommonExtractorTest.php]: clever!

**Comment by odinuv** [comment]
@pivnicek how about now?

**Comment by pivnicek** [comment]
@odinuv Great!  thank you, I wasn't sure how to use the test handler, that will be very useful in other cases too, wonderful.

**Pull request merged**